baseapp/abci.go

```diff
@@ -990,7 +990,7 @@ func (app *BaseApp) Commit() (*abci.ResponseCommit, error) {
 // disk in the Commit phase. This means when the ABCI client requests Commit(), the application
 // state transitions will be flushed to disk and as a result, but we already have
 // an application Merkle root.
-func (app *BaseApp) workingHash() []byte {
+func (app *BaseApp) WorkingHash() []byte {
```

Comment:
We shouldn't make this one public, if calling this solves the issue then the problem is that there are operations running outside of a FinalizeBlock call context. Because this function is called at the end of FinalizeBlock.